### ISSUE TITLE
Delete received private messages

### ISSUE THREAD

**Issue by aeharding** [LemmyNet/lemmy#4120]
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Is your proposal related to a problem?

Users should be able to delete received private messages. Right now they can only delete private messages they sent.

This is comment functionality available in Facebook Messenger, Reddit, phone text messaging etc

### Describe the solution you'd like.

Users can delete any private message from their inbox

### Describe alternatives you've considered.

There's no real alternative. I guess you could delete your account.

### Additional context

https://github.com/LemmyNet/lemmy/issues/3629#issuecomment-1778051833

**Comment by aeharding** [comment]
@MV-GH Most messaging apps I know of only delete the message from the inbox of the person that deleted it.

**Comment by Nutomic** [comment]
This is easy if sender and recipient are on different instances as we could simply delete the pm from the db. But if they are on the same instance it should still remain visible for the sender. So I suppose we can add a column `deleted_by_recipient` and filter by that.

**Comment by ell1e** [comment]
Some people send literal gore or worse, so this seems like it would in some situations be a sorely missed feature. When you report things it may take a while to get acted upon, so there should be a way to immediately nuke it from your inbox. The report dialog probably should offer to delete it from your inbox right after as well.

**Comment by aeharding** [comment]
@ell1e a workaround, blocking the account should remove DMs from that user in your inbox immediately.

**Comment by KaKi87** [comment]
Missing this feature is insane. Please make it a priority.

**Comment by Nothing4You** [comment]
considering that the database now has a column to mark PMs as removed this should be rather simple to implement that for the recipient only, while the `removed` column could be ignored when showing it to the sender.
when this is implemented, it should also be ensured that the sender cannot see that a message has been removed, assuming this clumn receives dual use for recipient-removal and admin-removal (via ban + remove).

**Comment by ell1e** [comment]
There are many use cases where deletion is useful and blocking just isn't. For example if you receive something questionable and you report it and you aren't sure if it'll raise to the bar for action, but you have a gut feeling future more incriminating messages will arrive soon. In that situation, there's no reason to keep the old upsetting messages sitting in your inbox forever.

**Comment by ell1e** [comment]
Also, can't a block be undone? Depending on what the message is, this might cause legal trouble for the recipient. Some messages you should be able to make yourself unable to access again. I don't think I need to explain which kind.

**Comment by KaKi87** [comment]
> But if you really need to do that then blocking would work better already anyway.

That also means keeping a personal list of all the spam you're receiving, that's not cool. Plus, the user is likely already banned by the instance admin (in my case it was), yet that action didn't delete the DM.

Also, blocking doesn't delete the DM either, it only hides it, because removing the block makes the DM reappear.

**Comment by smorks** [comment]
i have this mostly working, need to do some more testing and should be able to push a PR for this soon.

**Comment by victorz** [comment]
Can't wait to see it ❤️

**Comment by smorks** [comment]
> i have this mostly working, need to do some more testing and should be able to push a PR for this soon.

Oh i completely forgot about this. Sorry! I'll see if I still have my branch and see about getting this into a PR.

From my vague memory of when I worked on this before, I think I was having issues testing it because the backend and ui were in a weird state at that point, but I think things should be better now?

**Comment by dessalines** [comment]
Yes testing should be working fine, let us know if you need any help (or if we need to improve dev documentation for how to do that)

**Comment by smorks** [comment]
ok, found my branch, and after some trials and tribulations i have this close to done. should hopefully be ready for a PR in the next few days.

**Comment by ell1e** [comment]
Yay, thank you so much!! 🎉
